lib/membrane/core/timer.ex

```diff
@@ -28,8 +28,12 @@ defmodule Membrane.Core.Timer do
   end
 
   @spec stop(t) :: :ok
-  def stop(timer) do
-    Process.cancel_timer(timer.timer_ref)
+  def stop(%__MODULE__{timer_ref: nil}) do
```

Comment:
```suggestion
  def stop(%__MODULE__{interval: :no_interval}) do
```

I think that will be more meaningful